src/behavior-considered-undefined.md

```diff
@@ -57,6 +57,11 @@ code.
     > **Note**: `rustc` achieves this with the unstable
     > `rustc_layout_scalar_valid_range_*` attributes.
 
+Note that uninitialized memory is also implicitly invalid for any type that has
+a restricted set of valid values.  In other words, the only cases in which
+reading uninitialized memory is permitted is inside `union`s, and between the
```

Comment:
You can probably just minimize the second part to just "padding". We also don't define "compound type" anywhere, so it's probably best not to use it. We do define "user defined type" for struct/enum/union though.

Reply:
It's all "types with fields" though that I mean by "compound type" -- so, including tuples and arrays.